server/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesIndexResponse.java

```diff
@@ -68,55 +67,49 @@ public void writeTo(StreamOutput out) throws IOException {
         }
     }
 
-    private record GroupByMappingHash(List<String> indices, String indexMappingHash, Map<String, IndexFieldCapabilities> responseMap)
-        implements
-            Writeable {
-        GroupByMappingHash(StreamInput in) throws IOException {
-            this(in.readStringCollectionAsList(), in.readString(), in.readMap(IndexFieldCapabilities::new));
-        }
-
-        @Override
-        public void writeTo(StreamOutput out) throws IOException {
-            out.writeStringCollection(indices);
-            out.writeString(indexMappingHash);
-            out.writeMap(responseMap, StreamOutput::writeWriteable);
-        }
-
-        Stream<FieldCapabilitiesIndexResponse> getResponses() {
-            return indices.stream().map(index -> new FieldCapabilitiesIndexResponse(index, indexMappingHash, responseMap, true));
-        }
-    }
-
     static List<FieldCapabilitiesIndexResponse> readList(StreamInput input) throws IOException {
         if (input.getTransportVersion().before(MAPPING_HASH_VERSION)) {
             return input.readCollectionAsList(FieldCapabilitiesIndexResponse::new);
         }
-        final List<FieldCapabilitiesIndexResponse> ungroupedList = input.readCollectionAsList(FieldCapabilitiesIndexResponse::new);
-        final List<GroupByMappingHash> groups = input.readCollectionAsList(GroupByMappingHash::new);
-        return Stream.concat(ungroupedList.stream(), groups.stream().flatMap(GroupByMappingHash::getResponses)).toList();
+        final int ungrouped = input.readVInt();
```

Comment:
I'm guessing that this is in effect the same StreamInput/Output code as before, so we don't need to bump transport versions?

Reply:
Yea this step is 100% wire compatible :)